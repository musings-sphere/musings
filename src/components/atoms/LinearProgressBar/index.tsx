// import react library
import { useEffect, useState } from 'react';
import { LinearProgress } from '@mui/material';
import { styled } from '@mui/material/styles';

interface Props {
	delay?: number;
}

const ColorLinearProgress = styled(LinearProgress)(({ theme }) => ({
	colorPrimary: {
		backgroundColor: theme.palette.alternate.main,
	},
	barColorPrimary: {
		backgroundColor: theme.palette.primary.main,
	},
}));

const LinearProgressBar = ({ delay = 1000 }: Props): JSX.Element | null => {
	const [completed, setCompleted] = useState<number>(0);
	const [visibility, setVisibility] = useState<boolean>(false);

	const progress = () => {
		setCompleted((oldCompleted) => {
			if (oldCompleted === 100) return 0;
			const diff = Math.random() * 10;
			return Math.min(oldCompleted + diff, 100);
		});
	};

	useEffect(() => {
		const timer = setInterval(progress, 500);
		// cleanup function
		return () => clearInterval(timer);
	}, []);

	useEffect(() => {
		const timer = setTimeout(() => setVisibility(true), delay);
		// cleanup function
		return () => clearTimeout(timer);
	}, [delay]);

	return visibility ? (
		<ColorLinearProgress variant="determinate" value={completed} />
	) : null;
};

export default LinearProgressBar;
