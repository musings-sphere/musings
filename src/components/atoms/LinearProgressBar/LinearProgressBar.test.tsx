// react libraries
import { render } from '@testing-library/react';

// component
import LinearProgressBar from './index';

describe('GeneralCardInfo component', () => {
	const props = {};

	it('should render correctly', () => {
		const { asFragment } = render(<LinearProgressBar {...props} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
