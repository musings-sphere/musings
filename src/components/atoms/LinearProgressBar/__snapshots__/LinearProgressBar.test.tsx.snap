// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralCardInfo component should render correctly 1`] = `<DocumentFragment />`;
